chia/full_node/mempool_manager.py

```diff
@@ -103,6 +117,84 @@ def compute_assert_height(
     return height
 
 
+def run_for_cost_and_additions(
+    coin_id: bytes32, puzzle_reveal: SerializedProgram, solution: SerializedProgram, max_cost: uint64
+) -> Tuple[uint64, List[Coin]]:
+    cost, conditions_program = puzzle_reveal.run_mempool_with_cost(max_cost, solution)
+    created_coins = []
+    for s in conditions_program.as_iter():
```

Comment:
as you pointed out on keybase; this function should also account for the cost of `CREATE_COIN`, but we should probably not account for the cost-per-byte of the puzzle. We'll soon have compression that would deduplicate any identical puzzle reveals in the block anyway, so that's probably not a significant enough saving to count

Reply:
Done.